### PR TITLE
feat: enable tags for research + remove tag categories

### PR DESCRIPTION
**Pull request by iSCJT**
PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Developer experience (improves developer workflows for contributing to the project)

## Description

This pr:

- Introduces tag display for research.
- Removes tag categories.  As tags should be global for all information / form types.

## Git Issues

Closes #

## Screenshots/Videos

_If useful, provide screenshot or capture to highlight main changes_

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

File: src/pages/common/SortFilterHeader/SortFilterHeader.tsx [hunk 1]
```diff
@@ -222,7 +222,10 @@ export const SortFilterHeader = ({
               currentStore.activeSorter !== ItemSortingOption.MostRelevant
             ) {
               currentStore.updateActiveSorter(ItemSortingOption.MostRelevant)
-              setSortState({ label: 'Most Relevant', value: 'MostRelevant' })
+              setSortState({
+                label: ItemSortingOption.MostRelevant,
```
Comment (evakill) [inline src/pages/common/SortFilterHeader/SortFilterHeader.tsx:226]: nice use of the enum for the value; the label should still be `'Most Relevant'` (with the space)

File: src/common/Tags/TagsSelect.tsx [hunk 2]
```diff
@@ -63,40 +61,18 @@ const TagsSelect = (props: IProps) => {
...
-        item.tags &&
-        Object.keys(item.tags).forEach(
-          (tag) => (tagCounts[tag] = (tagCounts[tag] || 0) + 1),
-        ),
-    )
-
-    return tagCounts
-  }
-
-  if (props.relevantTagsItems) {
```
Comment (evakill) [inline src/common/Tags/TagsSelect.tsx:81]: remove this from the `props` if we are no longer going to support it

File: src/pages/Research/Content/ResearchListItem.tsx [hunk 3]
```diff
@@ -73,15 +73,33 @@ const ResearchListItem = ({ item }: IProps) => {
...
+                    mb={2}
+                    sx={{
+                      fontSize: [3, 3, 4],
+                    }}
+                  >
+                    {item.title}
+                  </Heading>
+                  <Flex
+                    sx={{
+                      display: ['none', 'inline-block', 'inline-block'],
```
Comment (evakill) [inline src/pages/Research/Content/ResearchListItem.tsx:88]: what does passing an array to the `display` do? haven't seen this before

File: src/pages/Research/Content/ResearchDescription.tsx [hunk 4]
```diff
@@ -26,7 +27,7 @@ import type { IResearch } from 'src/models/research.models'
 import type { IUser } from 'src/models/user.models'
 
 interface IProps {
-  research: IResearch.ItemDB
+  research: IResearch.ItemDB & { taglist: any }
```
Comment (evakill) [inline src/pages/Research/Content/ResearchDescription.tsx:30]: **(non-blocking)** i see this pattern is used in howto; but would be nice to apply more specific typing to `taglist` and make it camelCase

**Review by evakill — COMMENTED** [review]
looking good

**Comment by evakill** [comment]
works well on preview 👍 
Screenshot 2024-01-10 at 2 29 08 PM
Screenshot 2024-01-10 at 2 29 02 PM
Screenshot 2024-01-10 at 2 28 56 PM

@davehakkens in case he wants to look

**Comment by davehakkens** [comment]
Nice. Good to see the site-wide tags!
One small note, design team has actually been exploring the option to replace the display of tags with the category, because:
1. How-to categories bring significant differences in content type you will see
2. The design can be more controlled (currently it can mess up if many long tags are added)


afbeelding



afbeelding


That said. Happy to merge this for now and change this later when the decision is final

File: src/pages/Research/Content/ResearchListItem.tsx (hunk 3 above)
Comment (iSCJT) [reply src/pages/Research/Content/ResearchListItem.tsx:88]: It's break points for small, medium, large screens.  Can use it with any property, quite useful :semi

File: src/pages/common/SortFilterHeader/SortFilterHeader.tsx (hunk 1 above)
Comment (iSCJT) [reply src/pages/common/SortFilterHeader/SortFilterHeader.tsx:226]: thanks for the spot

File: src/pages/Research/Content/ResearchListItem.tsx (hunk 3 above)
Comment (evakill) [reply src/pages/Research/Content/ResearchListItem.tsx:89]: oooh nice!

**Review by evakill — APPROVED** [review]
thanks for the updates!!

**Pull request merged**